test/e2e/reconciler/scheduling/controller_test.go

```diff
@@ -257,3 +258,63 @@ func TestScheduling(t *testing.T) {
 		return found, fmt.Sprintf("no %s annotation:\n%s", schedulingv1alpha1.PlacementAnnotationKey, ns.Annotations)
 	}, wait.ForeverTestTimeout, time.Millisecond*100)
 }
+
+// TestSchedulingWhenLocationIsMissing create placement at first when location is missing and created later.
+func TestSchedulingWhenLocationUpdate(t *testing.T) {
+	t.Parallel()
+	framework.Suite(t, "transparent-multi-cluster")
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	source := framework.SharedKcpServer(t)
+
+	orgPath, _ := framework.NewOrganizationFixture(t, source, framework.TODO_WithoutMultiShardSupport())
+	locationPath, _ := framework.NewWorkspaceFixture(t, source, orgPath, framework.TODO_WithoutMultiShardSupport())
+	userPath, userWorkspace := framework.NewWorkspaceFixture(t, source, orgPath, framework.TODO_WithoutMultiShardSupport())
+
+	kcpClusterClient, err := kcpclientset.NewForConfig(source.BaseConfig(t))
+	require.NoError(t, err)
+
+	// create a placement at first
+	newPlacement := &schedulingv1alpha1.Placement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "new-placement",
+		},
+		Spec: schedulingv1alpha1.PlacementSpec{
+			LocationSelectors: []metav1.LabelSelector{{}},
+			NamespaceSelector: &metav1.LabelSelector{},
+			LocationResource: schedulingv1alpha1.GroupVersionResource{
+				Group:    "workload.kcp.io",
+				Version:  "v1alpha1",
+				Resource: "synctargets",
+			},
+			LocationWorkspace: locationPath.String(),
+		},
+	}
+	_, err = kcpClusterClient.Cluster(userPath).SchedulingV1alpha1().Placements().Create(ctx, newPlacement, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Placement should turn to pending phase")
+	framework.Eventually(t, func() (bool, string) {
+		placement, err := kcpClusterClient.Cluster(userPath).SchedulingV1alpha1().Placements().Get(ctx, newPlacement.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("Failed to get placement: %v", err)
+		}
+
+		return placement.Status.Phase == schedulingv1alpha1.PlacementPending, ""
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	syncTargetName := "synctarget"
+	t.Logf("Creating a SyncTarget and syncer in %s", locationPath)
+	_ = framework.NewSyncerFixture(t, source, locationPath,
+		framework.WithExtraResources("services"),
+		framework.WithSyncTargetName(syncTargetName),
+		framework.WithSyncedUserWorkspaces(userWorkspace),
+	).CreateSyncTargetAndApplyToDownstream(t).StartSyncer(t)
```

Comment:
You don't need to use `StartSyncer()` here, which will start the whole syncer, and start effective syncing (rather heavy and invasive).
Instead you could just call the `StartAPIImporter().StartHearBeat()`.

Going 1 step further you could even call the `StartAPIImporter` first, and wait for the `APIImporterReady` condition , and test that the placement is not ready (because the Heartbeat is not set), and then call `StartHeartBeat`, after which the placement will finally be updated.

This also allows testing the case when the heartbeat stops (Syncer stopped or unreachable) and comes back, a how the placements follows.